scripts/deploy.js

```
@@ -0,0 +1,145 @@
+}
+
+// NOTE: thumbnail.svg is optional
+if (!fs.existsSync(thumbnailPath)) {
+  thumbnailExists = false;
+  console.warn('WARNING: no thumbnail.svg found for project');
+}
+
+// Get app version and definition properties
+const  { version } = packageJson;
+const { strings, lifecycle_events, presentation_properties, name } = definition;
+
+// Presigned URL declarations
+let iconUrl, thumbnailUrl, sourceUrl;
+
+// Make requests to API
+fetch(`${apiUrl}/users/login`, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      credentials: 'include',
+      body: JSON.stringify({
+    		username: userName,
+    		password: userPassword
+    	})
+}).then(response => response.json()
+).then(result => {
+  const apiToken = result.Token;
+  console.log('Logged in, publishing application');
+  return fetch(`${apiUrl}/applications/${appId}/actions/publish`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${apiToken}`
+    },
+    credentials: 'include',
+    body: JSON.stringify({
+      version,
+      author_id: authorId,
+      name,
+      presentation_properties,
+      strings})
+  })
+}).then(response => {
+    return response.json();
+}
+).then(response => {
+  const { deployment_id: deploymentId } = response;
+  if (!deploymentId) {
+    throw new Error(JSON.stringify(response));
+  }
+  iconUrl = response.icon_url;
+  sourceUrl = response.source_url;
+  thumbnailUrl = response.thumbnail_url;
+})
+.then(() => {
+  if (iconExists) {
+    return uploadPresentationFile(iconPath, iconUrl);
+  }
+  console.warn('WARNING: No icon was uploaded');
+  return;
+})
+.then(() => {
+  if (thumbnailExists) {
+    return uploadPresentationFile(thumbnailPath, thumbnailUrl);
+  }
+  console.warn('WARNING: No thumbnail was uploaded');
+  return;
+}).then(() => uploadPresentationFile(bundlePath, sourceUrl)
+).catch(err => {
+	console.log(err);
```

We should throw an error here.